src/libos/crates/vdso-time/ocalls/vdso-time-ocalls.c

```diff
@@ -5,39 +5,27 @@
 
 int vdso_ocall_get_vdso_info(
     unsigned long *vdso_addr,
-    long *hres_resolution,
-    long *coarse_resolution,
     char *release,
-    int release_len,
-    struct timespec *tss,
-    int tss_len) {
+    int release_len) {
+    // If AT_SYSINFO_EHDR isn't found, getauxval will return 0.
     *vdso_addr = getauxval(AT_SYSINFO_EHDR);
 
-    struct timespec tp;
-    if (!clock_getres(CLOCK_REALTIME, &tp)) {
-        *hres_resolution = tp.tv_nsec;
-    } else { *hres_resolution = 0; }
-    if (!clock_getres(CLOCK_REALTIME_COARSE, &tp)) {
-        *coarse_resolution = tp.tv_nsec;
-    } else { *coarse_resolution = 0; }
-
     struct utsname buf;
     int ret = uname(&buf);
+    // uname should always succeed here, since uname only fails when buf is not invalid.
     if (ret != 0) { return -1; }
 
     int buf_rel_len = strlen(buf.release);
     int len = buf_rel_len > release_len ? release_len : buf_rel_len;
     memcpy(release, buf.release, len);
```

Comment:
As a rule of thumb, variable-length C strings should always be null terminated. All fields in utsname are null-terminated.

Reply:
```
    strncpy(release, buf.release, release_len);
    release[release_len - 1] = '\0';
```